File: FangZhiHu/app/src/main/java/network/DownProgressListener.java
package network;

/**
 * Created by yuantongqin on 2017/8/9.
 */

public interface DownProgressListener<T> {

    /**
     * 开始下载
     */
     void onStart();

    /**
     * 完成下载
     */
     void onComplete();


    /**
     * 下载进度
     */
    void updateProgress(long readLength, long countLength);

    /**
     * 失败
     */
     void onError(Throwable e);

}

File: FangZhiHu/CommonLibKotlin/src/main/java/network/download/DownProgressListener.kt
package network.download

/**
 * Created by tongqinyuan on 2018/3/7.
 */
interface DownProgressListener<T> {

    /**
     * 开始下载
     */
    fun onStart()

    /**
     * 完成
     */
    fun  onComplete()


    /**
     * 下载进度
     */
    fun updateProgress(readLength: Long, countLength: Long)

    /**
     * 出错
     */
    fun onError()
}